src/app/join-code.ts

```diff
@@ -1,5 +1,6 @@
-import { firestore } from 'firebase';
 import { milliseconds } from './utils/time-utils';
+import * as firebase from 'firebase/app';
+import firestore = firebase.firestore;
```

Comment:
Incidentally, it looks like both 

```ts
import { firestore } from 'firebase/app';
```

and

```ts
import * as firebase from 'firebase/app';
import firestore = firebase.firestore;
```

compile to the same JavaScript:
(I'm running `npm run build` and looking at `dist/main-es2015.js`)

```js
/* harmony import */ var firebase_app__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(/*! firebase/app */ "./node_modules/firebase/app/dist/index.cjs.js");
/* harmony import */ var firebase_app__WEBPACK_IMPORTED_MODULE_3___default = /*#__PURE__*/__webpack_require__.n(firebase_app__WEBPACK_IMPORTED_MODULE_3__);




var firestore = firebase_app__WEBPACK_IMPORTED_MODULE_3__["firestore"];
```

(These three lines of javascript are to the module-loading step, the default-export-getting step, and the subnamespace-aliasing step, respectively 🙂)

Reply:
```ts
import { firestore } from 'firebase/app';
```
definitely seems nicer, I didn't know I could do that on this version of Firebase. 

But, [in Firebase 8](https://firebase.google.com/support/release-notes/js#version_800_-_october_26_2020), they change how its exported and we have to import it like this:
```ts
import firebase from 'firebase/app'
```

in which case we can't use 
```ts
import { firestore } from 'firebase/app'
```

but we can still use

```ts
import firebase from 'firebase/app';
import firestore = firebase.firestore;
```

Also I believe the import for `firebase` is really just getting the core Firebase SDK and all the types the actual module for firestore, auth, or whatever gets loaded using separate imports that in our case are usually done in AngularFire for us, eg:
```ts
import 'firebase/firestore';
```
from AngularFire's firestore service (those are exported as empty in their type definitions).

(before this PR we were importing from just `firebase` which was the dev bundle containing everything in one bundle, but we don't want all the extra stuff from that)

Reply:
Ah, okay, thanks! 🙂 Gosh darn breaking changes.

In that case, do you mind if I add a comment?

```suggestion
import * as firebase from 'firebase/app';

// In this context, the `import` keyword just extracts a sub-namespace
// (firestore) from its parent namespace (firebase).
// This way, we don't have to type out `firebase.firestore` every time.
import firestore = firebase.firestore;
```

Reply:
Actually, we're using this import style in a lot of places, and it would be hard to comment all of them. I think I'm fine to just leave it as it is! 🙂

Thanks for explaining that.